src/Compilers/Core/Portable/SpecialTypeExtensions.cs

```diff
@@ -57,6 +57,35 @@ public static bool IsBlittable(this SpecialType specialType)
             }
         }
 
+        /// <summary>
+        /// Checks if a type is an immutable value type primitive.
+        /// </summary>
+        public static bool IsImmutableValueTypePrimitive(this SpecialType specialType)
```

Comment:
@gafter, I don't feel strongly about it.  I removed the extension per your feedback.